src/cascadia/TerminalSettingsEditor/Profiles.cpp

```diff
@@ -291,6 +331,16 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _DeleteProfileHandlers(*this, *deleteProfileArgs);
     }
 
+    void ProfilePageNavigationState::CreateUnfocusedAppearance()
+    {
+        _Profile.CreateUnfocusedAppearance(_Schemes, _WindowRoot);
+    }
+
+    void ProfilePageNavigationState::DeleteUnfocusedAppearance()
+    {
+        _Profile.DeleteUnfocusedAppearance();
+    }
+
```

Comment:
In an effort to get closer to #9207, any chance you can make the `State` _not_ have this responsibility?

Can this be a responsibility of the `ProfileViewModel` instead since that has a reference to the `_Profile` as well?

Reply:
For reference, a short version of the philosophy behind #9207 is that `State` is _only_ used in `OnNavigatedTo`; it should not be stored as a part of the page. Instead, that should be the responsibility of the view model, which stores the state of the page (aka "the view").

In an world after #9207 is resolved, we may not even need `State`. I know we're not there yet, but this might be one of those steps we can take to get us closer to that reality.

Reply:
I really wanted `State` to not have this responsibility, but unfortunately the `ProfileViewModel` does not have the list of colour schemes nor the window root, both of which are needed by the `AppearanceViewModel` that is being created.